src/coordination/coordinator_instance.cpp

```diff
@@ -33,6 +32,11 @@ CoordinatorInstance::CoordinatorInstance()
     : thread_pool_{1},
       raft_state_(RaftState::MakeRaftState(
           [this]() {
+            if (raft_state_.IsLockOpened()) {
+              spdlog::error("Leader hasn't encountered healthy state, doing force reset of cluster.");
+              thread_pool_.AddTask([this]() { this->ForceResetCluster(); });
```

Comment:
will thread from the thread pool start immediately?

Reply:
Almost immediatelly, as soon as there is conditional variable notify on change